### PR TITLE
rpm: update to 4.16.0

### PR DESCRIPTION
**Pull request by glaulher**

### CODE REVIEW

File: srcpkgs/rpm/patches/fix-musl.patch [hunk 1]
```diff
@@ -29,3 +29,4 @@ index 992d9ac..5f43b57 100644
  
  #include <rpm/rpmlog.h>
  #include "rpmio/digest.h"
+
```
Comment (ericonr) [inline srcpkgs/rpm/patches/fix-musl.patch:32]: Remove this newline

File: srcpkgs/rpm/template [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-2.0-or-later"
 homepage="https://rpm.org"
 changelog="https://rpm.org/wiki/Releases/${version}.html"
-distfiles="http://ftp.rpm.org/releases/rpm-4.15.x/rpm-${version}.tar.bz2"
-checksum=ddef45f9601cd12042edfc9b6e37efcca32814e1e0f4bb8682d08144a3e2d230
-python_version=2 #unverified
+distfiles="http://ftp.rpm.org/releases/rpm-${version%.*}.x/rpm-${version}.tar.bz2"
+checksum=ca5974e9da2939afb422598818ef187385061889ba766166c4a3829c5ef8d411
+python_version=2
```
Comment (ericonr) [inline srcpkgs/rpm/template:19]: Is 2 correct? It's linked against python3...

Comment (glaulher) [reply srcpkgs/rpm/template:19]: deleting this line, the package does not compile.

Comment (ericonr) [reply srcpkgs/rpm/template:19]: No, I think you should be using `python_version=3`

**Comment by ericonr** [comment]
The patch you made is patching the wrong file, you need to apply

```diff
diff --git a/srcpkgs/rpm/patches/fix-musl-rpmdb.patch b/srcpkgs/rpm/patches/fix-musl-rpmdb.patch
index e1e48b3b0d..2c0317ee5a 100644
--- a/srcpkgs/rpm/patches/fix-musl-rpmdb.patch
+++ b/srcpkgs/rpm/patches/fix-musl-rpmdb.patch
@@ -1,12 +1,11 @@
---- rpmdb.c
-+++ rpmdb.c
-@@ -1,7 +1,9 @@
-#include "system.h"
-
+--- lib/rpmdb.c
++++ lib/rpmdb.c
+@@ -8,6 +8,8 @@
+ #include <utime.h>
+ #include <errno.h>
+ #include <dirent.h>
++#include <sys/stat.h>
 +#include <fcntl.h>
-+#include <sys / stat.h>
-#include <popt.h>
-#include <rpm/rpmcli.h>
-#include <rpm/rpmdb.h>
-#include "cliutils.h"
-#include "debug.h"
+ 
+ #ifndef	DYING	/* XXX already in "system.h" */
+ #include <fnmatch.h>
```

to your repository instead.

The final patch should look like

```
--- lib/rpmdb.c
+++ lib/rpmdb.c
@@ -8,6 +8,8 @@
 #include <utime.h>
 #include <errno.h>
 #include <dirent.h>
+#include <sys/stat.h>
+#include <fcntl.h>
 
 #ifndef	DYING	/* XXX already in "system.h" */
 #include <fnmatch.h>
```

Also, rebase your branch so the merge commit goes away.

**Comment by glaulher** [comment]
the package compiled in musl but took 2h6m46s

**Comment by ericonr** [comment]
Your local repository might have been out of date, which would have made it build the outdated dependencies. Do `git remote add upstream https://github.com/void-linux/void-packages.git` then `git pull upstream master --autostash`

**Comment by glaulher** [comment]
for some reason trying to rebase I pushed the push up incorrectly, how do I fix it?

**Comment by ericonr** [comment]
`git rebase upstream/master` should do it

**Comment by ahesford** [comment]
Deeper and deeper!

Try this from the root of your clone, while you are in the `rpm` branch:
```
git fetch --all --prune
cp srcpkgs/rpm/{template,patches/fix-musl-rpmdb.patch} .
git reset --hard upstream/master
mv fix-musl-rpmdb.patch srcpkgs/rpm/patches/
mv template srcpkgs/rpm/
git add srcpkgs/rpm
git commit -m 'rpm: update to 4.16.0.'
git push -f
```

**Comment by glaulher** [comment]
> Cada vez mais fundo!
> 
> Tente fazer isso a partir da raiz do seu clone, enquanto você está no `rpm`branch:
> 
> ```
> git fetch --all --prune
> cp srcpkgs/rpm/{template,patches/fix-musl-rpmdb.patch} .
> git reset --hard upstream/master
> mv fix-musl-rpmdb.patch srcpkgs/rpm/patches/
> mv template srcpkgs/rpm/
> git add srcpkgs/rpm
> git commit -m 'rpm: update to 4.16.0.'
> git push -f
> ```


thanks for your help and ericonr's help, 
I think it's ok now.

File: srcpkgs/rpm/template (hunk 2 above)
Comment (ericonr) [reply srcpkgs/rpm/template:19]: ping here

Comment (ahesford) [reply srcpkgs/rpm/template:19]: It seems that there is exactly one Python script installed by `rpm`, and it works with Python 3 (I can at least get the help text to display). Please change to `python_version=3` as @ericonr recommends.

File: srcpkgs/rpm/template [hunk 3]
```diff
@@ -1,22 +1,22 @@
 # Template file for 'rpm'
 pkgname=rpm
-version=4.15.1
-revision=4
+version=4.16.0
+revision=2
```
Comment (ahesford) [inline srcpkgs/rpm/template:4]: This should be `revision=1` when the version is new.

**Review by ahesford — CHANGES_REQUESTED** [review]

File: srcpkgs/rpm/template (hunk 3 above)
Comment (glaulher) [reply srcpkgs/rpm/template]: the version has been fixed, thanks.

**Comment by ericonr** [comment]
Você ainda tinha dois commits (pode ver aqui https://github.com/void-linux/void-packages/pull/26642/commits ).  Eu juntei os dois e aproveitei para atualizar para 4.16.1.2.

**Comment by glaulher** [comment]
thank you.

**Pull request closed without merging**